DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+﻿// ReSharper disable InconsistentNaming
+
+namespace DigitalLearningSolutions.Data.Migrations
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Transactions;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Migrations.Properties;
+    using FluentMigrator;
+    using Microsoft.Data.SqlClient;
+
+    [Migration(202204071530, TransactionBehavior.None)]
+    public class PopulateUsersTableFromAccountsTables : Migration
+    {
+        public override void Up()
+        {
+            IDbConnection connection = new SqlConnection(ConnectionString);
+
+            var options = new TransactionOptions
+            {
+                Timeout = new TimeSpan(0, 15, 0),
+            };
+            using var transactionScope = new TransactionScope(TransactionScopeOption.Required, options);
+
+            // 1. Delete from Users (this should be empty)
+            connection.Execute("Delete Users");
+
+            // 2. Copy AdminAccounts to Users table
+            connection.Execute(
+                @"INSERT INTO dbo.Users (
+                    PrimaryEmail,
+                    PasswordHash,
+                    FirstName,
+                    LastName,
+                    JobGroupID,
+                    ProfessionalRegistrationNumber,
+                    ProfileImage,
+                    Active,
+                    ResetPasswordID,
+                    TermsAgreed,
+                    FailedLoginCount,
+                    HasBeenPromptedForPrn,
+                    LearningHubAuthId,
+                    HasDismissedLhLoginWarning,
+                    EmailVerified,
+                    DetailsLastChecked
+                )
+                SELECT Email,
+                    Password_deprecated,
+                    Forename_deprecated,
+                    Surname_deprecated,
+                    10,
+                    NULL,
+                    ProfileImage_deprecated,
+                    Active,
+                    ResetPasswordID_deprecated,
+                    TCAgreed_deprecated,
+                    FailedLoginCount_deprecated,
+                    0,
+                    NULL,
+                    0,
+                    GETUTCDATE(),
+                    GETUTCDATE()

```

Comment:
The default value when running this migration is to try to set the DetailsLastChecked to now. When we transfer admin accounts, we can be sure they are correct, but then when we try to add a delegate to the user, it may then become null or stay populated if everything is fine